components/brave_wallet/common/zcash_utils.h

```diff
@@ -265,6 +265,51 @@ struct OrchardTreeState {
   std::vector<uint8_t> frontier;
 };
 
+class OrchardShardTreeDelegate {
```

Comment:
Is there a reason for us to have `OrchardShardTreeDelegate`, and `OrchardShardTreeDelegateImpl`? It wasn't very obvious to me what is the benefit of breaking these in two.

I have also noticed that `ShardTreeDelegate` goes on to encapsulate `OrchardShardTreeDelegate`, and then it wraps around the operations for `OrchardShardTreeDelegate`. *I may be missing something*, but I'm wondering if we could potentially simplify this layering a bit, as there's quite a bit of redundancy with classes.

Reply:
Should also be some comments here explaining what this class does

Reply:
We have a discussion regarding this with @bridiver in dm.
I'll give some context here too:
We have a rust layer which operates cxx types and we have rust wrapper layer which converts cxx types to non-cxx types.
We also need to pass some interface to the rust to receive calls from rust to the underlying database and return result.
ShardTreeDelegate is a rust-layer interface. Shard tree uses it to call to the rust wrapper layer.
/zcash/rust/orchard_shard_tree_impl.cc acts as a rust wrapper, it implements ShardTreeDelegate which is passed to the rust layer, receives calls from the rust code and converts types from cxx to non-cxx. It needs some interface to make calls to the database. Since database is placed in the another layer(browser) and we can't depend from rust-wrapper layer on browser layer, we have this OrchardShardTreeDelegate interface. OrchardShardTreeDelegateImpl is a browser layer implementation that uses ZCashOrchardStorage inside.

Reply:
I've removed OrchardShardTreeDelegate. ShardTreeDelegate is now removed to CxxOrchardShardTreeDelegate. So we have CxxOrchardShardTreeDelegate which receives reference to the ZCashOrchardStorage instance. ZCashOrchardStorage now in a separate target so both browser/zcash/rust, browser/internal and browser/zcash can depend on it.